models/games_models.js

```diff
@@ -36,5 +36,26 @@ exports.selectReviewById = (review_id) => {
 };
 
 
-};
-
+exports.insertComment = (review_id, body) => {
+
+    if(!body.username || !body.body) {
+        return Promise.reject({status: 400, msg: 'Bad Request'})
+    } else {
+        return db.query(`
+    SELECT * FROM reviews WHERE review_id = $1;`, [review_id])
+     .then((result) => {
+        if (result.rows.length === 0){
+            return Promise.reject({status: 404, msg: `review does not exist for id: ${review_id}`});
+        } else {
+            return db.query(`
+    INSERT INTO comments 
+    (author, body, review_id, created_at, votes ) 
+    VALUES 
+    ($1, $2, $3, $4, 0)
+    RETURNING *;`, [body.username, body.body, review_id, new Date().toLocaleDateString()]).then((result) => {
+
+        return result.rows[0];
+    })
+        }
+});
+}};
```

Comment:
A few notes here, we dont need to select from reviews to find out if it exists firsts. When we insert the comment with the review_id, this is a foreign key, if they key doesnt exist to make a link in the other table, SQL will throw an error and we can then handle that error.

Also, we dont need to give the created_at and votes fields, these will be added by SQL as they have DEFAULT properties in the seed function when we created the table.

Lastly, this is starting to look a lot like nested promises and i want us to move away from, please refactor so this is only in a single promise chain